### DISCUSSION TITLE
guardar último valor y ubicación del botón

### DISCUSSION THREAD

**Discussion by Darksea1220** [sebasvelasco353/FDP-2022_01#26] [Q&A]
hola profe, recuerdas que el campo para preguntar lo de la operación lo mando desde el javascript? es que como le agregué el botón entonces me aparece el campo debajo del botón (cuando se le da al aceptar) y tampoco sé cómo guardar esa información en el arreglo, recomendaciones porfa.

**Comment by sebasvelasco353** [comment]
Hola.
No entiendo bien a qué te refieres con lo del botón, podes poner un screenshot porfa para verlo y decirme que es lo que debería de estar pasando?

Y con lo del array, necesitas que el array donde vas a guardar las cosas sea global, una vez tengas ese arreglo dentro de la función donde vas a guardar las cosas creas un objeto, por ejemplo

`Var ejemplo = {
    Nombre: document.getElementById("el Id de tu input de nombre").value
    // aquí pones todos los otros campos.
}

arrayDondeGuardo.push(ejemplo)`

**Reply by Darksea1220** [reply]
![boton arriba](https://user-images.githubusercontent.com/98364095/153735808-fb74e0e4-8f6a-4d4e-89a0-7b5c8351a816.png)
lo del array no lo puedo hacer por eso, como el campo lo estoy poniendo desde el javascript no tiene id

**Reply by Darksea1220** [reply]
![boton](https://user-images.githubusercontent.com/98364095/153736181-aa766b48-240d-4490-9613-fdfbc049c751.png)
El botón ya lo pude poner abajo sacándolo del div, pero no sé si allí me guarde los registros
y como no tengo allí el intput de las operaciones previas no tiene id (por lo que la creo desde el js) y por lo mismo no sé cómo llamarla luego al array (¿o cuando la creo el input desde el js se crea con un id por default?)

**Reply by sebasvelasco353** [reply]
ah ok ya entiendo a que te refieres.
No hay problema con que el boton este por fuera en este caso porque vas a obtener los resultados usando el .value de los elementos, y a esto puedes acceder desde el JS sin problema al darle click al boton y ejecutar handleDatosPaciente().

Para darle un id al eemento que creaste puedes usar 
`
  g = document.createElement('select'); 
  g.setAttribute("id", "el_ID");
`
pero en este caso no lo necesitas, tu le pides el valor es al select, que en este caso ya le pusiste un id="nacionalidad" entonces accedes a el usando

`
  var nacionalidad = document.getElementByID("nacionaidad").value;
`

**Reply by Darksea1220** [reply]
ok, gracias, pero ahora el problema que me surgió es que el botón no sirve, cuando le doy al botón de guardar registro, no guarda los datos, no hace nada

**Reply by Darksea1220** [reply]
tengo un condicional que está así 
`let age=datospaciente[4]
if (age>49) {
    var label2=document.createElement("label")
    label2.innerHTML="¿Se ha realizado pruebas de colesterol y triglicéridos en los ultimos 6 meses?"
    var input2=document.createElement("input")
    input2.type="select"
    document.getElementById("container__content").appendChild(label2);
    document.getElementById("container__content").appendChild(input2);
    
} else {
    var label3=document.createElement("label")
    label3.innerHTML="¿Se ha realizado pruebas de azucar en los ultimos 3 meses?"
    var input3=document.createElement("input")
    input3.type="select"
    document.getElementById("container__content").appendChild(label3);
    document.getElementById("container__content").appendChild(input3);`
para que me abra el campo que necesito según la edad del paciente, pero cuando abro la página, no espera a que llene los campos y tampoco cambian los valores cuando lleno los campos y oprimo el botón.

**Reply by Darksea1220** [reply]
![botonnosirve](https://user-images.githubusercontent.com/98364095/153738312-5e5211db-0584-4c8e-a3c7-8128cc9e1be2.png)
apenas cargo la página, ejecuta sin esperar a que llene los campos y me bota el de si es menor de 50 aunque luego ponga otro valor

**Reply by Darksea1220** [reply]
![problema id](https://user-images.githubusercontent.com/98364095/153738545-94f5e076-36a6-4083-8ea5-4e46343cbb36.png)
traté de darle el id al campo de operaciones previas y me dio un error

**Comment by sebasvelasco353** [comment]
let age=datospaciente[4] <--- cual es la razon para este codigo? 

Puede estar dando ese error porque estas leyendo value de un elemento que no tiene valor todavia.
segun la consola el error esta en infopaciente.

**Reply by Darksea1220** [reply]
![reader](https://user-images.githubusercontent.com/98364095/153974450-87ffd508-80c6-4839-b7d5-814356c49302.png)
ve profe, trato de usar ese reader, pero igual, cuando lo agrego deja de funcionar lo que ya tenía, lo he tratado de acomodar de muchísimas formas pero no da resultado y los tutoriales tampoco dan frutos.

**Reply by Darksea1220** [reply]
digo, listener

**Reply by sebasvelasco353** [reply]
no se si sea pq le pusiste el eventListener del click a nombre, es decir a tu input y no al boton, puede ser tambien pq estas haciendo push a un arreglo que se llama registroTemperaturas y no estoy seguro si existe o no en tu archivo, no veo nada mas.

intenta poniendo console logs dentro del metodo probar y dentro de los bloques que ejecutaria el if, de esa forma te das cuenta lo que esta pasando, por ejemplo si pones un console.log() dentro del metodo y nunca aparece en tu consola ya sabes que el problema es que nunca esta ejecutando ese metodo y te ayuda a entender mas facil lo que esta pasando con el codigo al ejecutarlo.
